app/src/main/java/org/jellyfin/mobile/player/queue/QueueManager.kt

```diff
@@ -218,19 +229,12 @@ class QueueManager(
      * a [MergingMediaSource] containing the mentioned media stream and all external subtitle streams.
      */
     @CheckResult
-    private fun prepareStreams(source: JellyfinMediaSource): MediaSource {
-        if (source.isDownload) {
-            val downloadDao: DownloadDao = get()
-            val fileUri = runBlocking {
-                downloadDao.getMediaUri(source.id)
-            }
-            return prepareDownloadStreams(source, fileUri)
-        } else {
-            return prepareSeverStreams(source)
-        }
-    }
+    private fun prepareStreams(source: LocalJellyfinMediaSource) = get<DownloadDao>()
+        .let { runBlocking { it.get(source.id) } }
+        .let(::requireNotNull)
```

Comment:
why do we get the source again if we already have it

Reply:
Ah! Good catch! Will resolve asap. 👍🏻

Reply:
Okay then I'll merge this PR